src/hello_freertos.c

```diff
@@ -1,54 +1,213 @@
 /**
- * Copyright (c) 2022 Raspberry Pi (Trading) Ltd.
- *
- * SPDX-License-Identifier: BSD-3-Clause
+*Chase and Ayman 2024 Lab 10
  */
 
 #include <stdio.h>
-
 #include "FreeRTOS.h"
 #include "task.h"
-
 #include "pico/stdlib.h"
-#include "pico/multicore.h"
 #include "pico/cyw43_arch.h"
 
-int count = 0;
-bool on = false;
+#define OUT_PIN 1 // GPIO pin for the external LED
 
-#define MAIN_TASK_PRIORITY      ( tskIDLE_PRIORITY + 1UL )
-#define BLINK_TASK_PRIORITY     ( tskIDLE_PRIORITY + 2UL )
-#define MAIN_TASK_STACK_SIZE configMINIMAL_STACK_SIZE
-#define BLINK_TASK_STACK_SIZE configMINIMAL_STACK_SIZE
+// Task priority and stack size definitions
+#define BLINK_TASK_PRIORITY     ( tskIDLE_PRIORITY + 2UL ) // Blink task priority
+#define BLINK_TASK_PRIORITY2    ( tskIDLE_PRIORITY + 5UL ) // Blink task priority
+#define BLINK_TASK_STACK_SIZE configMINIMAL_STACK_SIZE     // Stack size for blink task
 
+// Global variables
+int count = 0;      // Counter to toggle LED
+bool on = false;    // Current LED state (on or off)
+
+// 1. Blink an LED using sleep_ms to delay iterations
 void blink_task(__unused void *params) {
```

Comment:
If we really want to compare sleep_ms vs. vTaskDelay, we should run this case without the overhead of FreeRTOS